internal/fields/validate.go

```diff
@@ -783,11 +791,22 @@ func (v *Validator) parseSingleElementValue(key string, definition FieldDefiniti
 			return fmt.Errorf("the IP %q is not one of the allowed test IPs (see: https://github.com/elastic/elastic-package/blob/main/internal/fields/_static/allowed_geo_ips.txt)", valStr)
 		}
 	// Groups should only contain nested fields, not single values.
-	case "group":
-		switch val.(type) {
+	case "group", "nested":
+		switch val := val.(type) {
 		case map[string]interface{}:
-			// TODO: This is probably an element from an array of objects,
+			// This is probably an element from an array of objects,
 			// even if not recommended, it should be validated.
+			if v.specVersion.LessThan(semver3_0_0) {
+				break
+			}
+			errs := v.validateMapElement(key, common.MapStr(val), doc)
+			if definition.Type == "group" {
+				errs = append(errs, arrayOfObjectsErr)
```

Comment:
I don't think these errors could be filtered as it is the code.

The methods that get the errors and filter are just for the errors from package-spec. This is in `internal/validation` package. For instance
https://github.com/elastic/elastic-package/blob/29ded9d537b0fc2f9f62b66859cbb607726741fc/internal/validation/validation.go#L26

Reply:
Removed by now.